nipype/pipeline/plugins/tests/test_sgelike.py

```diff
@@ -0,0 +1,45 @@
+from nipype.pipeline.plugins.base import SGELikeBatchManagerBase
+from nipype.interfaces.utility import Function
+import nipype.pipeline.engine as pe
+from os.path import join
+import os
+from glob import glob
+import pytest
+from mock import patch
+from tempfile import TemporaryDirectory
+import subprocess
+
+
+def crasher():
+    raise ValueError()
+
+    
+def submit_batchtask(self, scriptfile, node):    
+    self._pending[1] = node.output_dir()
+    subprocess.call(["bash", scriptfile])
+    return 1
+
+
+def is_pending(self, taskid):
+    return False
+
+
+@patch.object(SGELikeBatchManagerBase, '_submit_batchtask', new=submit_batchtask)
+@patch.object(SGELikeBatchManagerBase, '_is_pending', new=is_pending)
+def test_crashfile_creation():
```

Comment:
Let's use the `tmp_path` fixture. This will replace the `TemporaryDirectory` construct.

```suggestion
def test_crashfile_creation(tmp_path):
```